<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>winMain</class>
 <widget class="QMainWindow" name="winMain">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>742</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
background-color: white
}</string>
  </property>
  <widget class="QWidget" name="wCentral">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SetNoConstraint</enum>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="2" column="1">
     <widget class="QWidget" name="wRenderOpts" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QTabWidget" name="twRenderOpts">
         <property name="maximumSize">
          <size>
           <width>250</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tabRenderOptsWorld">
          <attribute name="title">
           <string>World</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QScrollArea" name="scarOpts">
             <property name="widgetResizable">
              <bool>true</bool>
             </property>
             <widget class="QWidget" name="scarwWorldOpts">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>224</width>
                <height>433</height>
               </rect>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <widget class="QLabel" name="lbRS">
                 <property name="text">
                  <string>Schwarzschild radius</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="ledRS">
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="lbBgTex">
                 <property name="text">
                  <string>Background texture</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rbBgFromPreset">
                 <property name="text">
                  <string>From preset</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="lbBgPreset1">
                 <property name="enabled">
                  <bool>true</bool>
                 </property>
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="minimumSize">
                  <size>
                   <width>0</width>
                   <height>50</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rbBgFromFile">
                 <property name="text">
                  <string>From file</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QWidget" name="wBgFromFile" native="true">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <layout class="QHBoxLayout" name="horizontalLayout">
                  <item>
                   <widget class="QLineEdit" name="ledBgPath"/>
                  </item>
                  <item>
                   <widget class="QPushButton" name="pbChooseBgPath">
                    <property name="maximumSize">
                     <size>
                      <width>30</width>
                      <height>16777215</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>...</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
               <item>
                <spacer name="spWorldOpts">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>322</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tabRenderOptsObject">
          <attribute name="title">
           <string>Object</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QStackedWidget" name="stwObjectOpts">
             <property name="currentIndex">
              <number>0</number>
             </property>
             <widget class="QWidget" name="pageRn">
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QScrollArea" name="scarRingOpts">
                 <property name="widgetResizable">
                  <bool>true</bool>
                 </property>
                 <widget class="QWidget" name="scarwRingOpts">
                  <property name="geometry">
                   <rect>
                    <x>0</x>
                    <y>0</y>
                    <width>224</width>
                    <height>424</height>
                   </rect>
                  </property>
                  <layout class="QVBoxLayout" name="verticalLayout_4">
                   <item>
                    <widget class="QLabel" name="lbRnNumPt">
                     <property name="text">
                      <string>Number of particles</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="ledRnNumPt"/>
                   </item>
                   <item>
                    <widget class="QLabel" name="lbRnNormVec">
                     <property name="text">
                      <string>Normal vector</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QWidget" name="wRnNormVec" native="true">
                     <layout class="QFormLayout" name="formLayout">
                      <property name="horizontalSpacing">
                       <number>6</number>
                      </property>
                      <property name="verticalSpacing">
                       <number>0</number>
                      </property>
                      <property name="leftMargin">
                       <number>0</number>
                      </property>
                      <property name="topMargin">
                       <number>0</number>
                      </property>
                      <property name="rightMargin">
                       <number>0</number>
                      </property>
                      <property name="bottomMargin">
                       <number>0</number>
                      </property>
                      <item row="1" column="0">
                       <widget class="QLabel" name="lbRnNormVecY">
                        <property name="text">
                         <string>y</string>
                        </property>
                       </widget>
                      </item>
                      <item row="1" column="1">
                       <widget class="QLineEdit" name="ledRnNormVecY"/>
                      </item>
                      <item row="2" column="0">
                       <widget class="QLabel" name="lbRnNormVecZ">
                        <property name="text">
                         <string>z</string>
                        </property>
                       </widget>
                      </item>
                      <item row="0" column="1">
                       <widget class="QLineEdit" name="ledRnNormVecX"/>
                      </item>
                      <item row="2" column="1">
                       <widget class="QLineEdit" name="ledRnNormVecZ"/>
                      </item>
                      <item row="0" column="0">
                       <widget class="QLabel" name="lbRnNormVecX">
                        <property name="text">
                         <string>x</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLabel" name="lbRnRadius">
                     <property name="text">
                      <string>Radius</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="ledRnRadius"/>
                   </item>
                   <item>
                    <widget class="QLabel" name="lbRnSize">
                     <property name="text">
                      <string>Size</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="ledRnSize"/>
                   </item>
                   <item>
                    <widget class="QLabel" name="lbRnColor">
                     <property name="text">
                      <string>Color</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QWidget" name="wRnColor" native="true">
                     <layout class="QHBoxLayout" name="horizontalLayout_2">
                      <item>
                       <widget class="QLabel" name="lbRnColorDisplay">
                        <property name="styleSheet">
                         <string notr="true">QLabel {
	background-color : red;
	border: 1px solid gray
}</string>
                        </property>
                        <property name="text">
                         <string/>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QPushButton" name="pbRnChooseColor">
                        <property name="maximumSize">
                         <size>
                          <width>30</width>
                          <height>16777215</height>
                         </size>
                        </property>
                        <property name="text">
                         <string>...</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                   </item>
                   <item>
                    <spacer name="sRn">
                     <property name="orientation">
                      <enum>Qt::Vertical</enum>
                     </property>
                     <property name="sizeHint" stdset="0">
                      <size>
                       <width>173</width>
                       <height>213</height>
                      </size>
                     </property>
                    </spacer>
                   </item>
                  </layout>
                 </widget>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="page_4">
              <layout class="QVBoxLayout" name="verticalLayout_6"/>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="renderWidget" name="wglRender"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>742</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <addaction name="actionUser_Preferences"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuOptions"/>
  </widget>
  <action name="actionUser_Preferences">
   <property name="text">
    <string>User Preferences</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save...</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>renderWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>renderwidget.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
